### PR TITLE
PROD-30063: Add test coverage for flexible group request to join

### PR DESCRIPTION
**Pull request by Kingdutch**
## Problem / Solution
This commits adds test to document the behaviour of the "Request to Join" feature for flexible groups after discussing the behaviour with our PO and Designer.

The join functionality is moved to a new folder under "groups" in the testing structure. The outsider view test was previously modified to have a bit of test code around request to join and direct joining. This was moved to their own test so that the feature for viewing is really only for viewing and the joining features are separate.

The join functionality is not placed under flexible because after the group migration we're only left with a single group type. The functionality for other group-module-based things are different features.

## Issue tracker
https://getopensocial.atlassian.net/browse/PROD-30063

## How to test
- [x] The tests give sufficient test coverage for how our feature works
- [x] The tests pass

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

File: tests/behat/features/capabilities/groups/join/groups-join-direct.feature [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+  Scenario: As anonymous user view a public group
+    Given groups with non-anonymous owner:
+      | label      | field_group_description | type           | langcode | field_flexible_group_visibility | field_group_allowed_join_method |
+      | Test group | Public visibility       | flexible_group | en       | public                          | direct                          |
+    And I am an anonymous user
+
+    When I am viewing the "stream" page of group "Test group"
+
+    Then I should see "Test group"
```
Comment (BiaInacio) [inline tests/behat/features/capabilities/groups/join/groups-join-direct.feature:12]: As this file is under join folder, here I believe you could also check the join method by checking if the text or button with the name "Join" is present on the page

File: tests/behat/features/capabilities/groups/join/groups-join-request-manage.feature [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    Given I am logged in as a user with the "verified" role
+    # Todo: Fix the role declaration.
+    And I am a member of "Test group" with the "flexible_group-group_manager" role
+
+    When I am viewing the "membership-requests" page of group "Test group"
+    And I press "Approve Membership"
+    And I press "Yes"
+
+    Then I should see the success message "Membership request approved"
+    And I should see "No membership requests available."
```
Comment (BiaInacio) [inline tests/behat/features/capabilities/groups/join/groups-join-request-manage.feature:27]: In my opinion, checking that there are no memberships requests anymore is not enough, I would add a validation to certify that the manager approval was executed, by, for example, check if the approved member is listed under members tab.

File: tests/behat/features/capabilities/groups/join/groups-join-request-manage.feature [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    # Todo: Fix the role declaration.
+    And I am a member of "Test group" with the "flexible_group-group_manager" role
+
+    When I am viewing the "membership-requests" page of group "Test group"
+    And I press "Toggle Dropdown"
+    And I click "Reject Membership"
+    And I press "Yes"
+
+    Then I should see the success message "Membership request rejected"
+    And I should see "No membership requests available."
```
Comment (BiaInacio) [inline tests/behat/features/capabilities/groups/join/groups-join-request-manage.feature:40]: In my opinion, checking that there are no memberships requests anymore is not enough, I would add a validation to certify that the manager rejection was executed, by, for example, check if the rejected member is NOT listed under members tab.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3972-sckr2ylhyac6hrivstfwbahiicwowwvv.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/668cfc397685b9c64b4de50c

**Comment by Kingdutch** [comment]
I've applied all the feedback from @BiaInacio. Since the tests are also passing I'm merging this :)

**Pull request merged**